// @cometchat/chat-uikit-angular/esm2020/CometChatMessageBubble/cometchat-message-bubble/cometchat-message-bubble.component.mjs
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import { MenuListStyle } from '@cometchat/uikit-elements';
import { CometChatTheme, MessageBubbleAlignment } from '@cometchat/uikit-resources';
import * as i0 from "@angular/core";
import * as i1 from "../../CometChatTheme.service";
import * as i2 from "@angular/common";
export class CometChatMessageBubbleComponent {
    constructor(ref, themeService) {
        this.ref = ref;
        this.themeService = themeService;
        this.messageBubbleStyle = {
            width: "100%",
            height: "auto",
            background: "",
            borderRadius: "12px",
            border: "none"
        };
        this.alignment = MessageBubbleAlignment.right;
        this.options = [];
        this.id = undefined;
        this.optionsStyle = {
            width: "",
            height: "",
            border: "1px solid #e8e8e8",
            borderRadius: "8px",
            submenuWidth: "100%",
            submenuHeight: "100%",
            submenuBorder: "1px solid #e8e8e8",
            submenuBorderRadius: "8px",
            moreIconTint: "grey"
        };
        this.moreIconURL = "assets/moreicon.svg";
        this.topMenuSize = 3;
        this.theme = new CometChatTheme({});
        this.uikitConstant = MessageBubbleAlignment;
        this.isHovering = false;
        this.wrapperStyle = () => {
            switch (this.alignment) {
                case MessageBubbleAlignment.right:
                    return {
                        display: "flex",
                        justifyContent: "flex-end"
                    };
                case MessageBubbleAlignment.left:
                    return {
                        display: "flex",
                        justifyContent: "flex-start"
                    };
                case MessageBubbleAlignment.center:
                    return {
                        display: "flex",
                        justifyContent: "center"
                    };
                default:
                    return {
                        display: "flex",
                        justifyContent: "center"
                    };
            }
        };
        this.bubbleStyle = () => {
            return {
                ...this.messageBubbleStyle,
                display: "flex",
                flexDirection: "column",
                alignItems: "flex-start"
            };
        };
        this.optionsStyles = () => {
            return {
                justifyContent: this.alignment == MessageBubbleAlignment.left ? "flex-start" : "flex-end",
                top: this.headerView && this.alignment == MessageBubbleAlignment.left ? "-4px" : "-24px",
                background: this.themeService.theme.palette.getBackground()
            };
        };
    }
    ngOnChanges(changes) {
    }
    ngOnInit() {
        this.optionsStyle = new MenuListStyle({
            border: `1px solid ${this.themeService.theme.palette.getAccent200()}`,
            borderRadius: "8px",
            background: this.themeService.theme.palette.getBackground(),
            submenuWidth: "100%",
            submenuHeight: "100%",
            submenuBorder: `1px solid ${this.themeService.theme.palette.getAccent200()}`,
            submenuBorderRadius: "8px",
            submenuBackground: this.themeService.theme.palette.getBackground(),
            moreIconTint: this.themeService.theme.palette.getAccent()
        });
    }
    /**
     * hide show menu options on hover
     * @param  {MouseEvent} event?
     */
    hideShowMenuOption(event) {
        this.isHovering = event?.type === "mouseenter";
        this.ref.detectChanges();
    }
    /**
     * @param  {any} event
     */
    onOptionClick(event) {
        const onClick = event?.detail?.data?.onClick;
        if (onClick) {
            onClick(this.id, event?.detail?.event);
        }
    }
    bubbleAlignmentStyle() {
        return {
            display: "flex",
            justifyContent: "flex-start",
            alignItems: this.alignment == MessageBubbleAlignment.left ? "flex-start" : "flex-end",
        };
    }
    titleStyle() {
        return {
            display: "flex",
            justifyContent: this.alignment == MessageBubbleAlignment.left ? "flex-start" : "flex-end",
            alignItems: "flex-start"
        };
    }
}
CometChatMessageBubbleComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: CometChatMessageBubbleComponent, deps: [{ token: i0.ChangeDetectorRef }, { token: i1.CometChatThemeService }], target: i0.ɵɵFactoryTarget.Component });
CometChatMessageBubbleComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.11", type: CometChatMessageBubbleComponent, selector: "cometchat-message-bubble", inputs: { messageBubbleStyle: "messageBubbleStyle", alignment: "alignment", options: "options", id: "id", leadingView: "leadingView", headerView: "headerView", replyView: "replyView", contentView: "contentView", threadView: "threadView", footerView: "footerView", bottomView: "bottomView", moreIconURL: "moreIconURL", topMenuSize: "topMenuSize" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"cc-message-bubble__wrapper\" [ngStyle]=\"wrapperStyle()\">\n  <div class=\"cc-message-bubble__container\" (mouseenter)=\"hideShowMenuOption($event)\"\n(mouseleave)=\"hideShowMenuOption($event)\">\n<div class=\"cc-message-bubble__avatar\" *ngIf=\"leadingView\">\n  <ng-container *ngTemplateOutlet=\"leadingView\">\n  </ng-container>\n</div>\n<div class=\"cc-message-bubble\" [ngStyle]=\"bubbleAlignmentStyle()\">\n  <div class=\"cc-message-bubble__title\" [ngStyle]=\"titleStyle()\" *ngIf=\"headerView\">\n<ng-container [ngTemplateOutlet]=\"headerView\">\n</ng-container>\n  </div>\n  <div class=\"cc-messageoptions\" *ngIf=\"options && options.length >= 1 && isHovering\" [ngStyle]=\"optionsStyles()\">\n<cometchat-menu-list [moreIconURL]=\"moreIconURL\" [topMenuSize]=\"topMenuSize\" [menuListStyle]=\"optionsStyle\" [data]=\"options\" (cc-menu-clicked)=\"onOptionClick($event)\">\n</cometchat-menu-list>\n  </div>\n  <div class=\"cc-message-bubble__content\" [ngStyle]=\"bubbleStyle()\">\n<ng-container *ngTemplateOutlet=\"replyView\">\n</ng-container>\n<ng-container *ngTemplateOutlet=\"contentView\">\n</ng-container>\n<ng-container *ngTemplateOutlet=\"bottomView\">\n</ng-container>\n<ng-container *ngTemplateOutlet=\"threadView\">\n</ng-container>\n<div>\n</div>\n  </div>\n  <ng-container *ngTemplateOutlet=\"footerView\">\n  </ng-container>\n</div>\n  </div>\n</div>\n", styles: ["*{box-sizing:border-box}.cc-message-bubble__wrapper{position:relative}.cc-message-bubble__container{padding-right:8px;border-radius:inherit;display:flex;height:-moz-fit-content;height:fit-content;width:-moz-fit-content;width:fit-content;max-width:65%}.cc-message-bubble{flex:1 1 0;display:flex;flex-direction:column;justify-content:flex-start;align-items:flex-end;min-width:0;padding:8px 2px;border-radius:inherit}.cc-message-bubble__avatar{position:relative;border-radius:inherit;display:flex;flex-direction:column;justify-content:flex-start;align-items:flex-end;padding:8px 4px;margin-top:5px}.cc-message-bubble__title:empty:before{display:none}.cc-message-bubble__title:empty:after{display:block;width:100%;height:20px;padding:0 8px}.cc-message-bubble__content{border-radius:8px;width:-moz-fit-content;width:fit-content;overflow:hidden}.cc-message-bubble__content .ng-star-inserted{border-radius:inherit}.cc-messageoptions{position:absolute;top:-4px;display:flex;align-items:center;justify-content:flex-end;z-index:1}\n"], directives: [{ type: i2.NgStyle, selector: "[ngStyle]", inputs: ["ngStyle"] }, { type: i2.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { type: i2.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: CometChatMessageBubbleComponent, decorators: [{
            type: Component,
            args: [{ selector: 'cometchat-message-bubble', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div class=\"cc-message-bubble__wrapper\" [ngStyle]=\"wrapperStyle()\">\n  <div class=\"cc-message-bubble__container\" (mouseenter)=\"hideShowMenuOption($event)\"\n(mouseleave)=\"hideShowMenuOption($event)\">\n<div class=\"cc-message-bubble__avatar\" *ngIf=\"leadingView\">\n  <ng-container *ngTemplateOutlet=\"leadingView\">\n  </ng-container>\n</div>\n<div class=\"cc-message-bubble\" [ngStyle]=\"bubbleAlignmentStyle()\">\n  <div class=\"cc-message-bubble__title\" [ngStyle]=\"titleStyle()\" *ngIf=\"headerView\">\n<ng-container [ngTemplateOutlet]=\"headerView\">\n</ng-container>\n  </div>\n  <div class=\"cc-messageoptions\" *ngIf=\"options && options.length >= 1 && isHovering\" [ngStyle]=\"optionsStyles()\">\n<cometchat-menu-list [moreIconURL]=\"moreIconURL\" [topMenuSize]=\"topMenuSize\" [menuListStyle]=\"optionsStyle\" [data]=\"options\" (cc-menu-clicked)=\"onOptionClick($event)\">\n</cometchat-menu-list>\n  </div>\n  <div class=\"cc-message-bubble__content\" [ngStyle]=\"bubbleStyle()\">\n<ng-container *ngTemplateOutlet=\"replyView\">\n</ng-container>\n<ng-container *ngTemplateOutlet=\"contentView\">\n</ng-container>\n<ng-container *ngTemplateOutlet=\"bottomView\">\n</ng-container>\n<ng-container *ngTemplateOutlet=\"threadView\">\n</ng-container>\n<div>\n</div>\n  </div>\n  <ng-container *ngTemplateOutlet=\"footerView\">\n  </ng-container>\n</div>\n  </div>\n</div>\n", styles: ["*{box-sizing:border-box}.cc-message-bubble__wrapper{position:relative}.cc-message-bubble__container{padding-right:8px;border-radius:inherit;display:flex;height:-moz-fit-content;height:fit-content;width:-moz-fit-content;width:fit-content;max-width:65%}.cc-message-bubble{flex:1 1 0;display:flex;flex-direction:column;justify-content:flex-start;align-items:flex-end;min-width:0;padding:8px 2px;border-radius:inherit}.cc-message-bubble__avatar{position:relative;border-radius:inherit;display:flex;flex-direction:column;justify-content:flex-start;align-items:flex-end;padding:8px 4px;margin-top:5px}.cc-message-bubble__title:empty:before{display:none}.cc-message-bubble__title:empty:after{display:block;width:100%;height:20px;padding:0 8px}.cc-message-bubble__content{border-radius:8px;width:-moz-fit-content;width:fit-content;overflow:hidden}.cc-message-bubble__content .ng-star-inserted{border-radius:inherit}.cc-messageoptions{position:absolute;top:-4px;display:flex;align-items:center;justify-content:flex-end;z-index:1}\n"] }]
        }], ctorParameters: function () { return [{ type: i0.ChangeDetectorRef }, { type: i1.CometChatThemeService }]; }, propDecorators: { messageBubbleStyle: [{
                type: Input
            }], alignment: [{
                type: Input
            }], options: [{
                type: Input
            }], id: [{
                type: Input
            }], leadingView: [{
                type: Input
            }], headerView: [{
                type: Input
            }], replyView: [{
                type: Input
            }], contentView: [{
                type: Input
            }], threadView: [{
                type: Input
            }], footerView: [{
                type: Input
            }], bottomView: [{
                type: Input
            }], moreIconURL: [{
                type: Input
            }], topMenuSize: [{
                type: Input
            }] } });
