// @cometchat/chat-uikit-angular/esm2020/CometChatMessageComposer/cometchat-message-composer.module.mjs
import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { CometChatMessageComposerComponent } from "./cometchat-message-composer/cometchat-message-composer.component";
import "@cometchat/uikit-elements";
import { AIAssistBotMessageList } from "../Shared/Views/AIAssistBotMessageList/aiassist-bot-message-list.module";
import * as i0 from "@angular/core";
export class CometChatMessageComposer {
}
CometChatMessageComposer.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: CometChatMessageComposer, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CometChatMessageComposer.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: CometChatMessageComposer, declarations: [CometChatMessageComposerComponent], imports: [CommonModule, AIAssistBotMessageList], exports: [CometChatMessageComposerComponent] });
CometChatMessageComposer.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: CometChatMessageComposer, imports: [[CommonModule, AIAssistBotMessageList]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: CometChatMessageComposer, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CometChatMessageComposerComponent],
                    imports: [CommonModule, AIAssistBotMessageList],
                    exports: [CometChatMessageComposerComponent],
                    schemas: [CUSTOM_ELEMENTS_SCHEMA],
                }]
        }] });
