// @cometchat/chat-uikit-angular/esm2020/Extensions/Reactions/ReactionExtensionDecorator.mjs
import { CometChatMessageOption, CometChatUIKitConstants, localize, fontHelper } from "@cometchat/uikit-resources";
import { DataSourceDecorator } from "../../Shared/Framework/DataSourceDecorator";
export class ReactionExtensionDecorator extends DataSourceDecorator {
    constructor(dataSource) {
        super(dataSource);
    }
    getCommonOptions(loggedInUser, messageObject, theme, group) {
        let options = super.getCommonOptions(loggedInUser, messageObject, theme, group);
        if (!this.checkIfOptionExist(options, CometChatUIKitConstants.MessageOption.reactToMessage)) {
            let newOption = new CometChatMessageOption({
                id: CometChatUIKitConstants.MessageOption.reactToMessage,
                title: localize("ADD_REACTION"),
                iconURL: "assets/Reactionsicon.svg",
                onClick: null,
                iconTint: theme.palette.getAccent600(),
                backgroundColor: "transparent",
                titleFont: fontHelper(theme.typography.subtitle1),
                titleColor: theme.palette.getAccent600()
            });
            options.push(newOption);
        }
        return options;
    }
    checkIfOptionExist(template, id) {
        return template.some(obj => obj.id === id);
    }
    getId() {
        return "reactions";
    }
}
